Lib/python/pyrun.swg

```diff
@@ -335,9 +335,19 @@ swig_varlink_setattr(PyObject *o, char *n, PyObject *p) {
   return res;
 }
 
+#if PY_VERSION_HEX >= 0x03090000 || (!defined(SWIGPYTHON_BUILTIN) && PY_VERSION_HEX >= 0x03040000)
```

Comment:
This might need to be a user-level setting, since e.g. a custom `tp_traverse` function would need to be adjusted for heap types.

Reply:
Isn't tp_traverse intended for C structures?  Here, the only data stored by the C struct is a dict, and that struct is not customizable by the user.